plugins/cache-ehcache/src/main/java/org/opensearch/cache/store/disk/EhcacheDiskCache.java

```diff
@@ -151,6 +151,23 @@ private EhcacheDiskCache(Builder<K, V> builder) {
         if (this.storagePath == null || this.storagePath.isBlank()) {
             throw new IllegalArgumentException("Storage path shouldn't be null or empty");
         }
+        // Delete all the previous disk cache related files/data. We don't persist data between process restart for
+        // now which is why need to do this. Clean up in case there was a non graceful restart and we had older disk
+        // cache data still lying around.
+        Path ehcacheDirectory = Paths.get(this.storagePath);
+        if (Files.exists(ehcacheDirectory)) {
+            try {
+                logger.info("Found older disk cache data lying around during initialization under path: {}", this.storagePath);
+                IOUtils.rm(ehcacheDirectory);
+            } catch (IOException e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "Failed to delete ehcache disk cache data under path: {} " + "during initial",
+                        this.storagePath
```

Comment:
Should we fail the initialization instead ? Otherwise we may end up leaking the older files. Will cache initialization failure prevent the OS process from starting up ?

Reply:
Yeah cache initialization failure will prevent OS process to come up. Also failing the cache initialization seems more disruptive to me, so thought it would be better to log error and proceed.

Reply:
How can we detect that this cache file is getting leaked indeed ? Is there any metric which can capture this ? Otherwise we can potentially end up filling up the disk. I am fine if we don't fail the setup given the failure here will still allow the cache to work. Does cache file name changes with each restart ?

Reply:
One way would be to store the disk cache data under nodeId path [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/env/NodeEnvironment.java#L124) which gets tracked under nodeStats and by diskThresholdDecider. As of now disk cache storage path is configurable, but we can provide a default path but still doesn't guarantee the user would use the same.

Or we write an explicit tracker to track total_bytes_consumed by disk cache. Cache file names doesn't changes(as far I have seen in testing) but the parent folder name changes after process restart.

Reply:
> One way would be to store the disk cache data under nodeId path [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/env/NodeEnvironment.java#L124) which gets tracked under nodeStats and by diskThresholdDecider. As of now disk cache storage path is configurable, but we can provide a default path but still doesn't guarantee the user would use the same.

We will need to make sure that `DiskThresholdDecider` is aware of total disk usage either by data or by other components like this cache. Don't fully recall but i think decider tracks the total disk usage and not just under that path ? If this is incorrect, then we need to make sure decider is aware of the disk usage similar to how FileCacheStats are exposed to decider for search nodes.

> Cache file names doesn't changes(as far I have seen in testing) but the parent folder name changes after process restart.

In that case, if we just consume the failure then we will end up leaving the older cache files. I think for simplicity probably we can just fail the process to startup on the cleanup failure, such that on failure one can take a look and take necessary actions before restarting the OS process ? wdyt ?

Reply:
>We will need to make sure that DiskThresholdDecider is aware of total disk usage either by data or by other components like this cache. Don't fully recall but i think decider tracks the total disk usage and not just under that path

DiskThresholdDecider seems to track disk usage under nodeId path where usually all the data is stored, verified by doing a code walkthrough. 
For example [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDecider.java#L222), it tries to get total disk usage for a nodeId, and seems to fill disk usage somewhere [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/InternalClusterInfoService.java#L413). File cache data seems to be somewhere inside this path.

As part of segmented cache change, I will change the logic to store the caching data somewhere inside this folder by default until unless the user wants to override it.

> I think for simplicity probably we can just fail the process to startup on the cleanup failure

Yeah I think so. I will make the changes then.

Reply:
@sohami Added changes to throw exception this case.

Reply:
Also why to make this path configurable ?

Reply:
At that time, it was kept configurable mostly to give user the freedom to choose this path.